### ISSUE TITLE
AppSync returning old data

### ISSUE THREAD

**Issue by julien-tamade** [aws-amplify/aws-sdk-ios#2027]
**Describe the bug**
I am building an app that has users go through different voice over sessions. I have a simple graphQL schema with a user model that keeps track of a users current session number amongst other things. After each session I mutate the user so the current session number increases. But when I attempt to query the user to get their current session number, appsync is returning the old data even though the dynamoDB table has the new data.

**Comment by palpatim** [comment]
@julien-tamade Sorry you're having problems with this. Could you give us some more detail on this issue so we can help troubleshoot? Our "New issue" template for Bugs gives the minimum info, but please feel free to add more context to help us work through this with you.

---

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Which AWS service(s) are affected?**

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment(please complete the following information):**
 - SDK Version: [e.g. 2.6.29]
 - Dependency Manager: [e.g. Cocoapods, Carthage]
 - Swift Version : [e.g. 4.0]

**Device Information (please complete the following information):**
 - Device: [e.g. iPhone6, Simulator]
 - iOS Version: [e.g. iOS 11.4]
 - Specific to simulators:

**Additional context**
Add any other context about the problem here.

**Comment by julien-tamade** [comment]
thanks for your response @palpatim. I have since realized that the issue was because the query was pulling from the cache instead of the online database, and the cache wasn't being updated to reflect the updates in the database. I solved the issue temporarily by forcing the client to pull only from the database. 

However, now i'm wondering if this is another, separate issue with the cache not being updated when i submit a mutation. Should it be updated when i send a mutation? Right now the cache has outdated data even though i am sending mutations fine
